### ISSUE TITLE
Stack trace when aborting fetching of artifact list

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#141]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/141)
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Oct 29, 2017, 08:27

When launching a command which has no justification whatsoever for downloading the remote artifact list, it is highly likely for the user to hit `^C` a few times while BuildStream is taking an eternity to do a useless thing.

In this case, a really ugly thing happens.

The first `^C` prints "Aborted!", but buildstream does not return control to the calling shell:

```
$ bst track base/base-system.bst
Loading:   001
Resolving: 001/001
Fetching artifact list from https://gnome7.codethink.co.uk/artifacts
^C
Aborted!
```

The second `^C` does this ugly stack trace:
```
^CError in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python3.5/multiprocessing/popen_fork.py", line 29, in poll
    pid, sts = os.waitpid(self.pid, flag)
KeyboardInterrupt
```

At this point, you might think you are rid of this... but a child process has been orphaned, it's still trying to talk to the artifact server, how stubborn !

```
$ ps axf

[...]
 7262 pts/2    Sl     0:00 /usr/bin/python3 /home/tristan/.local/bin/bst track base/base-system.bst
```

You wont notice the above if you're not looking for it... but it will come and surprise you seconds later, possibly while you are doing something else in your shell:

```
$ Process Process-1:
Traceback (most recent call last):
  File "/usr/lib/python3.5/multiprocessing/process.py", line 249, in _bootstrap
    self.run()
  File "/usr/lib/python3.5/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/codethink/GNOME/buildstream/buildstream/_artifactcache/ostreecache.py", line 286, in child_action
    q.put((True, _ostree.list_remote_refs(self.repo, remote=remote)))
  File "/codethink/GNOME/buildstream/buildstream/_ostree.py", line 354, in list_remote_refs
    _, refs = repo.remote_list_refs(remote)
KeyboardInterrupt

```

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Oct 30, 2017, 11:12

changed title from **Stack trace when aborting {-obnoxious -}fetching of artifact list** to **Stack trace when aborting fetching of artifact list**

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Oct 30, 2017, 11:12

Removing obnoxious from the title, this is much less obnoxious now that #140 is fixed.

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Jan 10, 2018, 16:32

mentioned in commit da6cc4fea16d6ac07300aec6509292fba373e0c9

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Jan 10, 2018, 17:16

mentioned in commit bcb7eddff4850042569275d8a3e8325748909eec

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Jan 10, 2018, 17:19

mentioned in merge request !220

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Jan 10, 2018, 17:25

mentioned in commit 40a6a9554824ee18c95be4a545a7f20a9cb25f8d

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Jan 10, 2018, 17:26

mentioned in commit 0bd2e3b1fa0df81e0f9b00eaee08a877b4499847

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @samthursfield]](https://gitlab.com/samthursfield) on Jan 11, 2018, 17:35

mentioned in commit cc462c70473d8336d889249bff89c17ab52de2c1

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @samthursfield]](https://gitlab.com/samthursfield) on Jan 11, 2018, 18:18

mentioned in commit 20d653de3c46aa66b70116de68b78487225b2f23

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Jan 11, 2018, 19:55

closed via merge request !166

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Jan 11, 2018, 19:55

closed via commit 20d653de3c46aa66b70116de68b78487225b2f23

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tlater]](https://gitlab.com/tlater) on Jan 25, 2018, 14:11

mentioned in commit 44535a9a5cfb5c2bc828bb99220ef9269385e61a
